src/pagination/components/Pagination.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+
+class Pagination extends React.Component{
+    constructor(props){
+        super(props);
+        this.state={
+            allDataRowsNumber: 0,
+            currentPage: 1,
+            dataPerPage: props.dataPerPage
+        }
+        this.handleClick = this.handleClick.bind(this);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        this.setState({ allDataRowsNumber: nextProps.allDataRowsNumber });  
+    }
+
+    handleClick(event) {
+        event.preventDefault()
+        this.setState({
+          currentPage: Number(event.target.id)
+        });
+        this.props.handleClick(event.target.id);
+    }
+
+    render(){
+        const { allDataRowsNumber , currentPage, dataPerPage } = this.state;
+
+        const pageNumbers = [];
+        for (let i = 1; i <= Math.ceil(allDataRowsNumber / dataPerPage); i++) {
+            pageNumbers.push(i);
+        }
```

Comment:
This logic can be extracted to a helper function outside of `render()`. Also, keep in mind that this computation will be performed on every render (so, any time both props and state change). If `currentPage` changes, this is just extra CPU cycles... a challenge presents itself! 😮